pennylane/devices/null_qubit.py

```diff
@@ -0,0 +1,284 @@
+
+        self._operation_calls[operation.name] += 1
+
+        wires = operation.wires
+        if operation in diagonal_in_z_basis:
+            return self._apply_diagonal_unitary(state, None, wires)
+        if len(wires) <= 2:
+            # Einsum is faster for small gates
+            return self._apply_unitary_einsum(state, None, wires)
+        return self._apply_unitary(state, None, wires)
+
+    def _apply_x(self, state, axes, **kwargs):
+        self._operation_calls["PauliX"] += 1
+
+    def _apply_y(self, state, axes, **kwargs):
+        self._operation_calls["PauliY"] += 1
+
+    def _apply_z(self, state, axes, **kwargs):
+        self._operation_calls["PauliZ"] += 1
+
+    def _apply_hadamard(self, state, axes, **kwargs):
+        self._operation_calls["Hadamard"] += 1
+
+    def _apply_s(self, state, axes, inverse=False):
+        self._operation_calls["S"] += 1
+
+    def _apply_t(self, state, axes, inverse=False):
+        self._operation_calls["T"] += 1
+
+    def _apply_sx(self, state, axes, inverse=False):
+        self._operation_calls["SX"] += 1
+
+    def _apply_cnot(self, state, axes, **kwargs):
+        self._operation_calls["CNOT"] += 1
+
+    def _apply_swap(self, state, axes, **kwargs):
+        self._operation_calls["SWAP"] += 1
+
+    def _apply_cz(self, state, axes, **kwargs):
+        self._operation_calls["CZ"] += 1
+
+    def _apply_toffoli(self, state, axes, **kwargs):
+        self._operation_calls["Toffoli"] += 1
+
+    def _apply_phase(self, state, axes, parameters, inverse=False):
+        pass
+
+    def expval(self, observable, shot_range=None, bin_size=None):
+        pass
+
+    def var(self, observable, shot_range=None, bin_size=None):
+        pass
+
+    @classmethod
+    def capabilities(cls):
+        capabilities = super().capabilities().copy()
+        capabilities.update(
+            model="qubit",
+            supports_inverse_operations=True,
+            supports_analytic_computation=True,
+            supports_broadcasting=True,
+            returns_state=True,
+            passthru_devices={
+                "tf": "null.qubit.tf",
+                "torch": "null.qubit.torch",
+                "autograd": "null.qubit.autograd",
+                "jax": "null.qubit.jax",
+            },
+        )
+        return capabilities
+
+    @staticmethod
+    def _create_basis_state(index):
+        return None
+
+    @property
+    def state(self):
+        return None
+
+    def density_matrix(self, wires):
+        return None
+
+    def _apply_state_vector(self, state, device_wires):
+        pass
+
+    def _apply_basis_state(self, state, wires):
+        pass
+
+    def _apply_unitary(self, state, mat, wires):
+        pass
+
+    def _apply_unitary_einsum(self, state, mat, wires):
+        pass
+
+    def _apply_diagonal_unitary(self, state, phases, wires):
+        pass
+
+    def reset(self):
+        pass
```

Comment:
Wouldn't it make sense to refresh `self._operation_calls` here?

Reply:
Done! Thank you for your suggestion.